### PR TITLE
Add `env delete sandbox` and `env delete scratch` commands.

### PR DESCRIPTION
**Pull request by mshanemc**
requires https://github.com/salesforcecli/sf-plugins-core/pull/41

### What does this PR do?
use the new sfdx flags

### What issues does this PR fix or reference?
@W-10766801@ is the flags WI https://github.com/salesforcecli/sf-plugins-core/pull/32

this is an update to @W-10657714@ for the delete functionality to use the new flags

### CODE REVIEW

File: src/commands/env/delete/sandbox.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      char: 'p',
+      description: messages.getMessage('flags.no-prompt.summary'),
+    }),
+  };
+
+  public async run(): Promise<SandboxDeleteResponse> {
+    const { flags } = await this.parse(EnvDeleteSandbox);
+    const org = flags['target-org'];
+
+    if (flags['no-prompt'] || (await confirm(messages.getMessage('prompt.confirm', [org.getUsername()])))) {
```
Comment (peternhale) [inline src/commands/env/delete/sandbox.ts:43]: @mshanemc should be using prompter instance available via `this.prompter`

File: src/commands/env/delete/scratch.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      char: 'p',
+      description: messages.getMessage('flags.no-prompt.summary'),
+    }),
+  };
+
+  public async run(): Promise<ScratchDeleteResponse> {
+    const { flags } = await this.parse(EnvDeleteScratch);
+    const org = flags['target-org'];
+
+    if (flags['no-prompt'] || (await confirm(messages.getMessage('prompt.confirm', [org.getUsername()])))) {
```
Comment (peternhale) [inline src/commands/env/delete/scratch.ts:38]: Same here `this.prompter`

**Review by peternhale — CHANGES_REQUESTED** [review]

File: src/commands/env/delete/sandbox.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
```
Comment (RodEsp) [inline src/commands/env/delete/sandbox.ts:13]:
```suggestion
/*
 * Copyright (c) 2021, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */
```

**Review by RodEsp — CHANGES_REQUESTED** [review]
Looks good overall 👍🏼
Just a little nitpick on some duplicated license text.

**Review by RodEsp — DISMISSED** [review]

**Review by peternhale — DISMISSED** [review]

File: tsconfig.json [hunk 4]
```diff
@@ -3,7 +3,11 @@
   "compilerOptions": {
     "outDir": "lib",
     "rootDir": "src",
-    "skipLibCheck": true
+    "skipLibCheck": true,
+    "baseUrl": ".",
+    "paths": {
+      "@oclif/core": ["node_modules/@oclif/core"]
+    }
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply tsconfig.json:10]: `error TS2742: The inferred type of 'flags' cannot be named without a reference to '@salesforce/sf-plugins-core/node_modules/@oclif/core/lib/interfaces'. This is likely not portable. A type annotation is necessary.`

TS gets confused without it, similar to https://github.com/salesforcecli/plugin-source/blob/main/tsconfig.json, when there's other oclif/core in modules (like sf-plugin-core) even when they're the same version.

Definitely helps when doing yarn link locally

[the remark this answers is not in the document]
Comment (mshanemc) [reply tsconfig.json:10]: happy to try without it once the sf-plugin-core get merged but you'll need it for testing

[the remark this answers is not in the document]
Comment (RodEsp) [reply tsconfig.json:10]: Weird. Yeah, let's try it without after the `sf-plugin-core` PR gets merged.

**Review by RodEsp — APPROVED** [review]

**Pull request merged**